doc/new-guidelines.md

```diff
@@ -0,0 +1,454 @@
+ (fn [multiaccount]
+   (:preferred-name multiaccount)))
+```
+
+### Requiring quo2 components
+
+Never directly import a submodule of `quo2`, instead, use the provided
+`quo2.core` namespace.
+
+```clojure
+;; bad
+(ns my-namespace
+  (:require [quo2.components.icon :as icon]))
+
+(icon/icon :i/verified)
+
+;; good
+(ns my-namespace
+  (:require [quo2.core :as quo2]))
+
+(quo2/icon :i/verified)
+```
+
+### Require/import
+
+Prefer `:as` instead of `:refer`. There are exceptions to this rule, e.g. the
+test macros `deftest` and `is`, which are ubiquitous in the Clojure community.
+
+```clojure
+;; bad
+(ns status-im.utils.datetime
+  (:require [cljs-time.coerce :refer [from-long]]))
+
+;; good
+(ns status-im.utils.datetime
+  (:require [cljs-time.coerce :as time.coerce]))
+```
+
+### Accessibility labels
+
+Use keywords instead of strings. As a bonus, remember keywords are cached in
+memory.
+
+```clojure
+;; bad
+[text/text {:accessibility-label "profile-nickname"}
+ "Markov"]
+
+;; good
+[text/text {:accessibility-label :profile-nickname}
+ "Markov"]
+```
+
+### Icons
+
+Use the appropriate keyword qualification/namespace and don't directly require
+namespaces inside `quo2/components/`.
+
+```clojure
+;; bad
+(require '[quo2.components.icon :as icons])
+(icons/icon :main-icons2/verified)
+
+;; good
+(require '[quo2.core :as quo2])
+(quo2/icon :i/verified)
+```
+
+### Translations
+
+Prefer to use translation placeholders instead of creating multiple translation
+keywords and concatenating them into a single string.
+
+```clojure
+;; bad
+;; Assume the translation key is:
+;;   "biometric-auth-error": "Unable perform biometric authentication"
+(str (i18n/label :t/biometric-auth-error) "(" error-code ")")
+
+;; good
+;; Assume the translation key is:
+;;   "biometric-auth-error": "Unable perform biometric authentication ({{code}})"
+(i18n/label :t/biometric-auth-error {:code error-code})
+```
+
+## Project Structure
+
+As the mobile application grows in size and complexity, new patterns emerge to
+organize the codebase. The structure presented below is just an example of what
+the codebase could look like, so not all pieces are presented here.
+
+```
+src
+├── common
+│  └── bottom_sheet.cljs
+├── contexts
+│   ├── browser/
+│   ├── communities
+│   │   ├── components/
+│   │   ├── screens
+│   │   │   └── __test__
+│   │   └── utils
+│   │       └── example.cljs
+│   ├── messaging/
+│   ├── user_settings/
+│   └── wallet/
+├── quo2
+│   ├── components/
+│   ├── foundations/
+│   ├── screens/
+│   └── theme.cljs
+├── react_native
+│   ├── gesture.cljs
+│   └── platform.cljs
+├── setup
+│   ├── core.cljs
+│   ├── events.cljs
+│   ├── navigation.cljs
+│   ├── reloader.cljs
+│   └── test_runner.cljs
+├── subs
+│   ├── ens.cljs
+│   └── pairing.cljs
+└── utils
+    ├── number.cljs
+    └── test_helpers.cljs
+```
+
+### Example structure for a quo2 component
+
+Notice the `test.cljs` file does not live inside a `__tests__/` directory, as
```

Comment:
I found this part slightly confusing. I think we just remove `__tests__` part from this example and describe how we handle tests in a separate section

Reply:
Cool, I'm going to give it another spin @J-Son89. The hardest part is always explaining the project structure and the rationale, because the text easily grows in size and becomes confusing.